docs/tutorial/reading_writing.md

```diff
@@ -0,0 +1,336 @@
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = path.read.flatMap(file => IO(println(file)))
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/readme.txt")
+
+  def run: IO[Unit] = FilesOs.read(path).flatMap(file => IO(println(file)))
+
+end App
+```
+
+@:@
+
+Congratulations! You have just loaded the contents of a file in pure Scala 🎉.
+
+### Exercise
+
+You may not like to keep using the `flatMap` function over and over again to sequence computations. This is why there is the [macro keyword `for`](https://docs.scala-lang.org/scala3/book/control-structures.html#for-expressions) to automatically let the compiler write the `flatMap`s for you. Why don't you try rewriting the program we just did using for-comprehensions? 
+
+```scala
+def run: IO[Unit] =
+  for
+    // Complete your code here!
+    ...
+  yield ()
+```
+[See solution](https://gist.github.com/f8cf952fd5f7341ef698dfe825cfca93.git)
+
+## Writing and modifying the contents of a file
+
+Now that you know how to load a file, you might also want to modify it and save it.
+
+To write to a file, use the `write` function with one of its variants so that you can save it to a completely different file. Here, we reverse the file so it reads backwards (if that is something you want to do):
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/change_me.txt")
+
+  def run: IO[Unit] =
+    for
+      file <- path.read
+      reversedFile = file.reverse
+      _ <- path.write(reversedFile)
+    yield ()
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/change_me.txt")
+
+  def run: IO[Unit] =
+    for
+      file <- FilesOs.read(path)
+      reversedFile = file.reverse
+      _ <- FilesOs.write(path, reversedFile)
+    yield ()
+
+end App
+```
+
+@:@
+
+Be aware that this will overwrite the contents of the file. So be careful not to change important files while you are learning!
+
+### Exercise
+
+Try loading the contents of two different files, concatenating them, and saving the result to a third location. How would you do it?
+
+[See possible solution](https://gist.github.com/99880af9b051b15f8adace7fe7aee2e3.git)
+
+## Working line by line
+
+There is a useful variant of the standalone methods called `Lines` (e.g. `readLines`), which reads the file line by line and stores them on a `List[String]`. This comes handy when you are working with a list of things that you want to convert:
```

Comment:
```suggestion
Shellfish provides a method called `readLines`, which reads the file line by line and stores them on a `List[String]`. This comes handy when you are working with a list of things that you want to convert:
```
the sentence reads a bit confusing to me, standalone methods of what? I don't think you have to mention there are variants, if `readLines` will be the default.

Reply:
I mention that on the wiki section :)